server/src/main/java/org/activityinfo/server/endpoint/rest/DatabaseResource.java

```diff
@@ -147,4 +172,245 @@ public Response createFormFromXForm(@Context UriInfo uri, XForm xForm) {
                 .build())
                 .build();
     }
+
+    private Database getDatabase() {
+        return entityManagerProvider.get().find(Database.class, databaseId);
+    }
+
+    private Maybe<User> getUserByEmail(String userEmail) throws NoResultException {
+        User user;
+        try {
+            user = entityManagerProvider.get().createQuery(
+                    "SELECT u FROM User u " +
+                            "WHERE u.email = :email", User.class)
+                    .setParameter("email", userEmail)
+                    .getSingleResult();
+            return Maybe.of(user);
+        } catch (NoResultException e) {
+            return Maybe.notFound();
+        }
+    }
+
+    @POST
+    @Path("/transfer/request")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response requestTransfer(@InjectParam AuthenticatedUser executingUser, String json) {
```

Comment:
The JAX-RS framework handles serialization/deserialization of requests and responses. The signature of this method should be:

```
public Response startTransfer(@InjectParam AuthenticatedUser executingUser, RequestTransfer request);

```

Just make sure RequestTransfer implements JsonSerializable.

Reply:
Great, didnt know that. Will change these signatures to reflect this.